Cannot find module for a peerDependency of a devDependency

### pnpm version: 0.38.1

### Code to reproduce the issue:
```
npm i -g angular-cli
ng new myproject --skip-npm
cd myproject
pnpm i
ng build
```
EDIT:
Faster way to reproduce, thanks to @zkochan :
```
git clone https://github.com/zkochan/pnpm-issue394.git
cd pnpm-issue394
ng build
```

### Expected behavior:
A folder dist is created in the project directory containing the app to deploy.

### Actual behavior:
The command fails with the error: **Cannot find module 'typescript'**

### Additional information:
```
Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\aecz\.pnpm-store\@ngtools+webpack@1.0.0\_\src\plugin.js:2:10)
```
Typescript dependency is correctly linked in myproject/node_modules but @ngtools/webpack do not "see" it and declares it as a peerDependency. Maybe because it is itself a dependency of angular-cli. Note that angular-cli is also correctly linked locally as the project declares it as devDependency.

Node v6.5.0 on Windows 7

If you dont have a global store, then another error appear indeed:
```
Error: Cannot find module '@angular/tsc-wrapped/src/symbols'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\aecz\pnpm-issue394\node_modules\.store\@angular-cli+ast-tools@1.0.2\_\src\ast-utils.js:4:17)
```
Although it works fine using npm install :(
It seems that a sub dependency does not see its siblings. With npm3 and the flat tree this issue is solved.

--- a/src/fs/relSymlink.ts
+++ b/src/fs/relSymlink.ts
@@ -1,6 +1,8 @@
 import symlink, {SymlinkType} from './forceSymlink'
 import path = require('path')
 import os = require('os')
+import mkdirp from './mkdirp'
+import normalize = require('normalize-path')
 
 // Always use "junctions" on Windows. Even though support for "symbolic links" was added in Vista+, users by default
 // lack permission to create them
@@ -9,9 +11,17 @@ const symlinkType: SymlinkType = os.platform() === 'win32' ? 'junction' : 'dir'
 /**
  * Relative symlink
  */
-export default function relSymlink (src: string, dest: string) {
+export default async function relSymlink (src: string, dest: string) {
   // Junction points can't be relative
   const rel = symlinkType !== 'junction' ? path.relative(path.dirname(dest), src) : src
 
-  return symlink(rel, dest, symlinkType)
+  try {
+    await symlink(rel, dest, symlinkType)
+  } catch (err) {
+    if ((<NodeJS.ErrnoException>err).code === 'ENOENT') {
+      await mkdirp(normalize(dest).split('/').slice(0, -1).join('/'))
+      await symlink(rel, dest, symlinkType)
+      return
+    }
+  }
 }

--- a/src/install/index.ts
+++ b/src/install/index.ts
@@ -44,7 +44,7 @@ export type InstalledPackage = {
   optional: boolean,
   id: string,
   keypath: string[],
-  escapedName: string,
+  name: string,
   fromCache: boolean
 }
 
@@ -120,7 +120,7 @@ export default async function install (ctx: InstallContext, pkgMeta: PackageMeta
         optional,
         keypath,
         id: freshPkg.id,
-        escapedName: spec.escapedName,
+        name: spec.name,
         fromCache: false
       }
       log('package.json', pkg)
@@ -177,7 +177,7 @@ export default async function install (ctx: InstallContext, pkgMeta: PackageMeta
         id: path.basename(fullpath),
         optional,
         keypath,
-        escapedName: spec.escapedName,
+        name: spec.name,
         fromCache: true
       }
     }
@@ -261,13 +261,16 @@ async function symlinkSelf (target: string, pkg: Package, depth: number) {
     return
   }
   await mkdirp(path.join(target, 'node_modules'))
+  const src = isScoped(pkg.name)
+    ? path.join('..', '..', '_')
+    : path.join('..', '_')
   await relSymlink(
-    path.join('..', '_'),
-    path.join(target, 'node_modules', escapeName(pkg.name)))
+    src,
+    path.join(target, 'node_modules', pkg.name))
 }
 
-function escapeName (name: string) {
-  return name && name.replace('/', '%2f')
+function isScoped (pkgName: string): boolean {
+  return pkgName.indexOf('/') !== -1
 }
 
 /**

--- a/src/install/linkPeers.ts
+++ b/src/install/linkPeers.ts
@@ -43,9 +43,9 @@ export default async function linkPeers (store: string, installs: InstalledPacka
   }))
 
   await Promise.all(Object.keys(peers).map(async function (name) {
-    await unsymlink(path.join(modules, peers[name].escapedName))
+    await unsymlink(path.join(modules, peers[name].name))
     return relSymlink(
       path.join(store, peers[name].id, '_'),
-      path.join(modules, peers[name].escapedName))
+      path.join(modules, peers[name].name))
   }))
 }

